esphome/components/xiaomi_cgpr1/binary_sensor.py

```diff
@@ -0,0 +1,75 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor, binary_sensor, esp32_ble_tracker
+from esphome.const import (
+    CONF_BATTERY_LEVEL,
+    CONF_BINDKEY,
+    CONF_DEVICE_CLASS,
+    CONF_MAC_ADDRESS,
+    CONF_ID,
+    DEVICE_CLASS_EMPTY,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ILLUMINANCE,
+    ICON_EMPTY,
+    UNIT_PERCENT,
+    CONF_IDLE_TIME,
+    CONF_ILLUMINANCE,
+    UNIT_MINUTE,
+    UNIT_LUX,
+    ICON_TIMELAPSE,
+)
+
+DEPENDENCIES = ["esp32_ble_tracker"]
+AUTO_LOAD = ["xiaomi_ble"]
```

Comment:
This allows for not requiring an empty `sensor` key to be added to yaml
```suggestion
AUTO_LOAD = ["xiaomi_ble", "sensor"]
```